### PR TITLE
Fix syntax analysis for compiler directives

### PR DESCRIPTION
**Pull request by temanbrcom**
Compiler options processing moved to the main parser to allow precise syntax analysis for them. 
The technical grammar is extracted not to specify all the tokens from the ComonCobolLexer to allow them as copybook names.

### CODE REVIEW

**Comment by asatklichov** [comment]
How applyLookAhead behaves now?   Before there were I guess 4 tests activated it.

File: server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CICSParser.g4 [hunk 1]
```diff
@@ -927,11 +979,11 @@ cics_value: LPARENCHAR ptr_value RPARENCHAR;
 empty_parens: LPARENCHAR RPARENCHAR;
 
 cobolWord
-   : IDENTIFIER | cics_only_words | db2sql_only_words
+   : IDENTIFIER | cics_only_words | db2sql_only_words | cobolCompilerDirectivesKeywords
```
Comment (asatklichov) [inline server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CICSParser.g4:982]: In future, we should clean-up / separate these parsers as well (e.g. CICSParser misleading with non-related cics rules)

**Review by asatklichov — APPROVED** [review]
☕

File: server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CommonCobolLexer.g4 [hunk 2]
```diff
@@ -207,145 +200,54 @@ VOLUME : V O L U M E;
...
-fragment R:('r'|'R');
-fragment S:('s'|'S');
-fragment T:('t'|'T');
-fragment U:('u'|'U');
-fragment V:('v'|'V');
-fragment W:('w'|'W');
-fragment X:('x'|'X');
-fragment Y:('y'|'Y');
-fragment Z:('z'|'Z');
+C_CHAR: C {!sqlFlag}?;
```
Comment (grianbrcom) [inline server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CommonCobolLexer.g4:217]: This looks alarming...

File: server/src/main/resources/org/eclipse/lsp/cobol/core/strategy/SpecialTokenHandling.properties [hunk 3]
```diff
@@ -21,5 +21,17 @@ COMMENTTAG=
 COMMENTENTRYTAG=
 COMMENTENTRYLINE=
 COMMENTLINE=
+C_CHAR=C
```
Comment (grianbrcom) [inline server/src/main/resources/org/eclipse/lsp/cobol/core/strategy/SpecialTokenHandling.properties:24]: How about `AR`, `CO` and others?

File: server/src/test/java/org/eclipse/lsp/cobol/usecases/TestOneCharCompilerOptions.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2020 Broadcom.
```
Comment (grianbrcom) [inline server/src/test/java/org/eclipse/lsp/cobol/usecases/TestOneCharCompilerOptions.java:2]: Change your template.

**Review by grianbrcom — APPROVED** [review]
:grapes:

File: server/src/main/resources/org/eclipse/lsp/cobol/core/strategy/SpecialTokenHandling.properties (hunk 3 above)
Comment (temanbrcom) [reply server/src/main/resources/org/eclipse/lsp/cobol/core/strategy/SpecialTokenHandling.properties:24]: No need, this in only for tokens whose names don't match the assigned patterns

File: server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CommonCobolLexer.g4 (hunk 2 above)
Comment (temanbrcom) [reply server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CommonCobolLexer.g4:217]: Agree, it looks terrible. Separating the parser should make it easier

**Pull request merged**